### PR TITLE
🛑 DON'T MERGE [test] Fix source-stability test and revert #18793

### PR DESCRIPTION
**Pull request by lorentey**
rdar://problem/43425867

### CODE REVIEW

**Comment by lorentey** [comment]
@swift-ci please test

**Review by aschwaighofer — APPROVED** [review]

File: test/api-digester/source-stability.swift.expected [hunk 1]
```diff
@@ -1,4 +1,6 @@
 /* FIXME: Bogus */
+Struct DictionaryIterator has generic signature change from <Key, Value where Key : Hashable> to <Element>
+Struct SetIterator has generic signature change from <Element where Element : Hashable> to <Key, Value>
```
Comment (jrose-apple) [inline test/api-digester/source-stability.swift.expected]: Wait, this seems backwards? Why would they switch like this?

**Comment by lorentey** [comment]
@swift-ci test

File: test/api-digester/source-stability.swift.expected (hunk 1 above)
Comment (nkcsgexi) [reply test/api-digester/source-stability.swift.expected]: The correct diags are actually like these:
-Struct DictionaryIterator has generic signature change from <Key, Value where Key : Hashable> to <Element>
-Struct SetIterator has generic signature change from <Element where Element : Hashable> to <Key, Value>
+Struct DictionaryIterator has generic signature change from <Key, Value where Key : Hashable> to <Key, Value>
+Struct SetIterator has generic signature change from <Element where Element : Hashable> to <Element>

Comment (lorentey) [reply test/api-digester/source-stability.swift.expected]: I copy & pasted the ones in this PR from my own test logs, with no edits -- is it possible the api-digester output is somehow nondeterministic?

**Comment by lorentey** [comment]
We'll see a second run soon; the new CI build is almost done compiling.

**Comment by lorentey** [comment]
Yep, the test has passed in the current build, despite the earlier failure:
```
PASS: Swift(macosx-x86_64) :: api-digester/source-stability.swift (7913 of 11307)
```

**Comment by nkcsgexi** [comment]
Thanks, I'll investigate what's going on.

**Comment by nkcsgexi** [comment]
hmm, i cannot reproduce the nondeterministic output issue locally. We should run the swift-ci multiple times here to see if it reproduces.

**Comment by nkcsgexi** [comment]
oh, i can reproduce this issue with more runs. It seems `SetIterator` and `DictionaryIterator` have been changed from a type decl to a typealias decl, thus the tool is confused with mapping. I'll investigate the fix.

**Comment by lorentey** [comment]
See #18811 for the correct fix.

**Pull request closed without merging**